igvc_description/urdf/jessi.urdf.xacro

```diff
@@ -0,0 +1,460 @@
+              <kd>10</kd>
+              <max_vel>0.01</max_vel>
+              <min_depth>0.005</min_depth>
+            </ode>
+          </contact>
+        </surface>
+      </collision>
+      <inertial>
+        <origin xyz="0 0 0" />
+        <mass value="${wheel_mass}"/>
+        <inertia  ixx="${1 / 12 * wheel_mass * wheel_width**2 + 1/4 * wheel_mass * (tire_diameter / 2)**2}"
+                  ixy="0.0"
+                  ixz="0.0"
+                  iyy="${1 / 12 * wheel_mass * wheel_width**2 + 1/4 * wheel_mass * (tire_diameter / 2)**2}"
+                  iyz="0.0"
+                  izz="${1/2 * wheel_mass * (tire_diameter / 2)**2}" />
+      </inertial>
+    </link>
+
+    <xacro:drive_shock lr_prefix="${lr_prefix}" lr_reflect="${lr_reflect}"/>
+
+    <gazebo reference="${lr_prefix}_wheel">
+      <mu1>0.7</mu1>
+      <mu2>0.75</mu2>
+    </gazebo>
+
+  </xacro:macro>
+
+  <!-- macro used to define casters -->
+  <xacro:macro name="caster" params="ball_radius ball_mass tire_diameter x y">
+    <link name="caster">
+      <inertial>
+        <mass value="${ball_mass}" />
+        <origin xyz="0 0 0" rpy="0 0 0"/>
+        <inertia  ixx="${2 / 5 * ball_mass * ball_radius ** 2}" ixy="0.0" ixz="0.0"
+                  iyy="${2 / 5 * ball_mass * ball_radius ** 2}" iyz="0.0"
+                  izz="${2 / 5 * ball_mass * ball_radius ** 2}" />
+      </inertial>
+      <visual>
+        <geometry>
+          <sphere radius="${ball_radius}"/>
+        </geometry>
+      </visual>
+      <collision>
+        <geometry>
+          <sphere radius="${ball_radius}"/>
+        </geometry>
+      </collision>
+    </link>
+
+    <joint name="base_link_to_caster" type="floating">
+      <parent link="base_link"/>
+      <child link="caster"/>
+      <origin xyz="${x} ${y} ${ball_radius - tire_diameter / 2}" rpy="0 0 0"/>
+    </joint>
+
+    <gazebo reference="caster">
+      <mu1>0.2</mu1>
+      <mu2>0.25</mu2>
+    </gazebo>
+
+    <gazebo>
+      <joint name='base_link_to_caster' type='ball'>
+        <parent>base_link</parent>
+        <child>caster</child>
+        <pose frame=''>0 0 0 0 0 0</pose>
+        <physics>
+          <ode>
+            <limit>
+              <cfm>0</cfm>
+              <erp>0.2</erp>
+            </limit>
+            <suspension>
+              <cfm>0</cfm>
+              <erp>0.2</erp>
+            </suspension>
+          </ode>
+        </physics>
+      </joint>
+    </gazebo>
+  </xacro:macro>
+
+
+  <link name="center_cam">
+      <visual>
+          <geometry>
+              <box size="0.1 0.1 0.1"/>
+          </geometry>
+      </visual>
+  </link>
+
+  <joint name="base_link_to_center_cam" type="fixed">
+      <origin xyz="${camera_x} ${camera_y} ${camera_height}" rpy="0 ${camera_pitch} 0" />
+      <parent link="base_link"/>
+      <child link="center_cam"/>
+  </joint>
+
+  <!-- the optical joint is in the correct frame for transforms -->
+  <link name="optical_center_cam"/>
```

Comment:
Was the choice to use a prefix instead of the usual suffix "_optical" here intentional?
ref: http://www.ros.org/reps/rep-0103.html#suffix-frames